### PR TITLE
Allow `getSubscribedEvents` to return a Generator

### PR DESCRIPTION
**Pull request by ruudk**
| Q             | A
| ------------- | ---
| Bug fix?      |no
| New feature?  | yes
| Doc updated   |no
| BC breaks?    |no 
| Deprecations? |no 
| Tests pass?   | yes
| Fixed tickets | #832 
| License       | Apache-2.0

### CODE REVIEW

**Comment by docteurklein** [comment]
when would it make sense to use generators in this method ?

File: tests/Serializer/EventDispatcher/EventDispatcherTest.php [hunk 1]
```diff
@@ -156,6 +156,25 @@ public function testAddSubscriber()
...
+
+        $this->dispatcher->addSubscriber($subscriber);
+        $this->assertAttributeEquals(array(
+            'foo.bar_baz' => array(
+                array(array($subscriber, 'onfoobarbaz'), null, 'foo'),
+            ),
+            'bar' => array(
+                array(array($subscriber, 'bar'), 'foo', null),
+            ),
+        ), 'listeners', $this->dispatcher);
```
Comment (docteurklein) [inline tests/Serializer/EventDispatcher/EventDispatcherTest.php]: we shouldn't assert the internal state of the SUT I guess.

File: src/JMS/Serializer/EventDispatcher/EventSubscriberInterface.php [hunk 2]
```diff
@@ -32,7 +32,7 @@
      * The class may be omitted if the class wants to subscribe to events of all classes.
      * Same goes for the format key.
      *
-     * @return array
+     * @return array|\Generator
```
Comment (docteurklein) [inline src/JMS/Serializer/EventDispatcher/EventSubscriberInterface.php]: wouldn't the pseudotype `iterable` work here?

**Comment by goetas** [comment]
in reality with symfony, this gets compiled, so there is no really a reason to use generators here...

**Comment by goetas** [comment]
closing

**Pull request closed without merging**